src/proxy/inbound.rs

```diff
@@ -114,10 +123,24 @@ impl Inbound {
     /// handle_inbound serves an inbound connection with a target address `addr`.
     pub(super) async fn handle_inbound(
         request_type: InboundConnect,
+        origin_src: Option<IpAddr>,
         addr: SocketAddr,
     ) -> Result<(), std::io::Error> {
         let start = Instant::now();
-        let stream = TcpStream::connect(addr).await;
+        let stream = {
+            match &request_type {
+                DirectPath(_) => super::freebind_connect(origin_src, addr).await,
+                Hbone(req) => {
+                    let origin_src = if origin_src.is_some() {
```

Comment:
why only if it is_some()?

Reply:
answer myself: None is if its not enabled; we always get original src if its enabled